python/qemu/ot/rom/image.py

```diff
@@ -238,6 +309,7 @@ def subst_perm_enc(cls, in_, key, width, num_round):
 
     @classmethod
     def subst_perm_dec(cls, val, key, width, num_round):
```

Comment:
Types on this function signature like with `sub_perm_enc`?
```suggestion
    def subst_perm_dec(cls, val: int, key: int, width: int, num_round: int) -> int:
```